Nil pointer in cluster autoscaler simulator in 0.3.0-beta5

cc: @piosz @fgrzadkowski @jszczepkowski 

```
I0902 18:04:57.444449       5 cluster.go:189] Looking for place for default/nginx-3358184317-bjw2p
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x52b8a8]
goroutine 36 [running]:
panic(0x1b648a0, 0xc820010070)
        /usr/local/google/home/mwielgus/.gvm/gos/go1.6.2/src/runtime/panic.go:481 +0x3e6
k8s.io/contrib/cluster-autoscaler/simulator.findPlaceFor.func2(0xc82097b200, 0x1a, 0xc82094d900, 0x1a)
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/simulator/cluster.go:160 +0x778
k8s.io/contrib/cluster-autoscaler/simulator.findPlaceFor(0xc820ab8a40, 0x1a, 0xc8200209b8, 0x1, 0x1, 0xc820c413f0, 0x13, 0x16, 0xc82096db00, 0xc
820020370, ...)
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/simulator/cluster.go:193 +0x833
k8s.io/contrib/cluster-autoscaler/simulator.FindNodesToRemove(0xc82086ad00, 0x13, 0x20, 0xc820c413f0, 0x13, 0x16, 0xc820a00400, 0x6d, 0x80, 0x0,
 ...)
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/simulator/cluster.go:95 +0x977
main.FindUnneededNodes(0xc820c413f0, 0x13, 0x16, 0xc820ad1140, 0x3fe0000000000000, 0xc820a00400, 0x6d, 0x80, 0xc820020370, 0xc820c201e0, ...)
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/scale_down.go:85 +0x363
main.run(0xc8203613e0)
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/cluster_autoscaler.go:283 +0x2360
created by k8s.io/contrib/cluster-autoscaler/vendor/k8s.io/kubernetes/pkg/client/leaderelection.(*LeaderElector).Run
        /usr/local/google/home/mwielgus/ca-0.3/src/k8s.io/contrib/cluster-autoscaler/vendor/k8s.io/kubernetes/pkg/client/leaderelection/leaderel
ection.go:177 +0x91
```

--- a/cluster-autoscaler/simulator/cluster.go
+++ b/cluster-autoscaler/simulator/cluster.go
@@ -157,6 +157,13 @@ func findPlaceFor(removedNode string, pods []*kube_api.Pod, nodes []*kube_api.No
 			nodeInfo, found = nodeInfos[nodename]
 		}
 		if found {
+			if nodeInfo.Node() == nil {
+				// NodeInfo is generated based on pods. It is possible that node is removed from
+				// an api server faster than the pod that were running on them. In such a case
+				// we have to skip this nodeInfo. It should go away pretty soon.
+				glog.Warningf("No node in nodeInfo %s -> %v", nodename, nodeInfo)
+				return false
+			}
 			nodeInfo.Node().Status.Allocatable = nodeInfo.Node().Status.Capacity
 			err := predicateChecker.CheckPredicates(pod, nodeInfo)
 			glog.V(4).Infof("Evaluation %s for %s/%s -> %v", nodename, pod.Namespace, pod.Name, err)

